### ISSUE TITLE
`Graph API`: structural query fails on boolean parameter comparison

### ISSUE THREAD

**Issue by indietyp** [hashintel/hash#1666]
### Describe the bug

The Graph API returns an internal server error when comparing two boolean parameters in the filter (to return all entities). This is not the case for the comparison of string parameters.

```json
{
  "filter": {
    "equal": [{"parameter": true}, {"parameter": true}]
  },
  "graphResolveDepths": {
    "inheritsFrom": {
      "outgoing": 0
    },
    "constrainsValuesOn": {
      "outgoing": 0
    },
    "constrainsPropertiesOn": {
      "outgoing": 0
    },
    "constrainsLinksOn": {
      "outgoing": 0
    },
    "constrainsLinkDestinationsOn": {
      "outgoing": 0
    },
    "isOfType": {
      "outgoing": 0
    },
    "hasLeftEntity": {
      "outgoing": 0,
      "incoming": 0
    },
    "hasRightEntity": {
      "outgoing": 0,
      "incoming": 0
    }
  }
}
```

### Error Log

```log
external-services-graph-1                |   2022-12-14T12:47:31.015627Z ERROR graph::api::rest::entity: Could not read entities from the store, error: Could not query from store
external-services-graph-1                | ├╴ /usr/local/src/hash_graph/lib/graph/src/store/postgres/knowledge/entity/read.rs:81:14
external-services-graph-1                | │
external-services-graph-1                | ╰─▶ error serializing parameter 0: cannot convert between the Rust type `bool` and the Postgres type `text`
external-services-graph-1                |     ├╴ /usr/local/src/hash_graph/lib/graph/src/store/postgres/knowledge/entity/read.rs:80:14
external-services-graph-1                |     ╰╴ backtrace with 205 frames (1)
```

### To reproduce

1. `POST /entites/query` with the payload described above	

### Expected behavior

Returns all entities

**Comment by TimDiekmann** [comment]
Thanks for filing this!

This is an interesting find and I'll look into this.

In order to return all entities (or ontology types for the other query endpoints), you could also use
```json
{
  "filter": {
    "all": []
  }
}
```

This will be slightly faster as well (but most probably not noticeable at all). An empty `"all"` filter will always evaluate to `true`, an empty `"any"` filter will always evaluate to `false.

**Comment by indietyp** [comment]
fixed in #1965
